### ISSUE TITLE
Error showing when logging in with anonymous

### ISSUE THREAD

**Issue by editter** [bp2008/ui3#97]
When loading the page using anonymous authentication I get an error stating "Blue Iris sent session data instead of an authentication challenge (probably indicates a Blue Iris bug)".  It appears that it expects the first request of login to fail and then the second attempt actually performs the login.
In the function LogInWithCredentials my response is a response.result == "success" and response.data.reason == undefined which causes the code to go to the else block however I don't think that is valid in my case.

My proposed solution would be to add an else if check inside LogInWithCredentials that checks for success. Thoughts?
```javascript
if (response && response.result == "fail" && response.data && response.data.reason == "missing response")
{
   ... 
   // existing code stays the same
   ...
} else if(response.result == "success"){
   // the server says its a valid attempt so use the data it provided
   self.HandleSuccessfulLogin(response, isAutomatic)
} else
   onFail(response, 'Blue Iris sent session data instead of an authentication challenge (probably indicates a Blue Iris bug): ' + JSON.stringify(args) + " -> " + JSON.stringify(response));
```

**Comment by bp2008** [comment]
What BI version?  Can you screenshot the BI web server advanced settings?  I'd like to reproduce the problem so I can see if it is something that needs fixed in BI.

**Comment by editter** [comment]
Also, note that I think the issue is because I had user/password settings saved in the "Upgrade Session" dialog. When I removed those credentials and reloaded the page I didn't get the error so when the page loads it appears to be trying to elevate the user which is good, however, that first attempt is expected to fail and doesn't so maybe my proposed solution isn't a good option.  Side note, should the "Upgrade Session" dialog even appear if Anonymous has admin privileges?

Steps to reproduce:
- Setup BI to use anonymous auth
- In UI3 right-click a camera and click "Restart Camera"
- When prompted with the "Upgrade Session" dialog enter the credentials of a non-Anonymous admin user/password and click Yes to "Log in automatically".
- Click "Log in"
- Refresh the page and see an error

BI Version: 5.5.3.3 x64
![image](https://user-images.githubusercontent.com/4632327/143080590-a7810dec-aa74-4a0a-89db-c302c03dcf64.png)
![image](https://user-images.githubusercontent.com/4632327/143082201-aaf8138c-437f-4f68-af55-20f863055bcf.png)

The response from ExecJson using args {cmd:"login"} when credentials are saved in "Upgrade Session" dialog 

```json
{
  "result": "success",
  "session": "[removed]",
  "data": {
    "system name": "CamServer",
    "admin": false,
    "changeprofile": true,
    "ptz": true,
    "audio": true,
    "clips": true,
    "clipcreate": true,
    "timelimits": false,
    "sessionlimit": 0,
    "streamlimit": 3600,
    "daylimit": 0,
    "dayused": 0,
    "dio": false,
    "version": "5.5.3.3",
    "support": "3/22/2022 12:00:35 AM",
    "user": "Anonymous",
    "latitude": [removed],
    "longitude": [removed],
    "tzone": "[removed]",
    "streams": [
      "CBR 512 kbps; gop 300, bframe 0",
      "CBR 1024 kbps; gop 300, bframe 0",
      "CBR 256 kbps; gop 300, bframe 0"
    ],
    "sounds": [
      ... all defaults removing to save space...
    ],
    "www_sounds": [
     ... all defaults removing to save space...
    ],
    "profiles": [
      "Inactive",
      "Active",
      "Work hours",
      "Profile 3",
      "Profile 4",
      "Profile 5",
      "Profile 6",
      "Profile 7"
    ],
    "schedules": [
      "Default"
    ],
    "news": false
  }
}
```

**Comment by bp2008** [comment]
I'm unable to reproduce the problem.  However I am testing BI 5.5.3.4 and I know that latest patch included some work on anonymous authentication (specifically it allows a new anonymous session to be created via the login API) so I suggest you try updating.

The first json API request UI3 makes is to test the current session to see what permissions it has, and see if it matches saved credentials so it knows whether to try a new login.  This request payload will be `{"cmd":"login","session":"6d8..."}` and it is expected that the server will return a success response with session data.

When I follow your steps, in the final step, after refreshing the page, UI3 discovers that the existing session already was authenticated using the saved credentials, and does not proceed to try another login.  If I log out and then load UI3 via bookmark such that I get an anonymous session, then UI3 notices the user name does not match what is saved, to UI3 performs a new login.  On my system it just works as intended.  On your system, the login fails because something has broken the login handshake protocol.

Normally, a login procedure goes like this:

1. Client sends request `{"cmd":"login"}`.
2. Server sends response `{"result":"fail","session":"08d...","data":{"auth-exempt":true,"reason":"missing response"}}` which includes a new session string.
3. Client does `MD5(user + ":" + session + ":" + pass)` and sends that in a new login command, response field: `{"cmd":"login","session":"08d...","response":"a1f..."}`
4. Server sends success response containing session data.

For the server to send a success response containing session data at step #2, that means the server thought you sent a session string.  It has happened in the past where BI would pull the session string from an HTTP cookie, but that was fixed shortly after, and it was a long time ago now.  Maybe BI 5.5.3.3 had a recurrence of that bug.  Anyway it is not happening to me in BI 5.5.3.4.

> Side note, should the "Upgrade Session" dialog even appear if Anonymous has admin privileges?

No, it should not appear.  It appears because Blue Iris is telling UI3 the anonymous session does not have admin privileges:

![image](https://user-images.githubusercontent.com/5639911/143095114-568fec9b-3115-4d62-a1e8-2857b443b243.png)

It might be that the admin permission is not being granted by Blue Iris.  Or may be the API response is just incorrect when it says admin: false.  Either way, that would be a bug in Blue Iris and you should report it to BI support.

I really have no idea why the login protocol is breaking for you.

**Comment by editter** [comment]
When updating to the latest BI my anonymous session connections jumped to the max allowed (I temporarily increased it to 75 and it took all of those as well but that may be a BI issue).

Interestingly enough I changed the user in BI from "Anonymous" to "anonymous" (notice the Capital A) and now things seem to be working fine.  I did also disable the Anonymous user altogether before the name change and was prompted with the default browser Basic Auth dialog.  Then re-enabled the user and changed the name and voila, now it seems like the BI json endpoint is working as expected.
